tools/SetupFlow/DevHome.SetupFlow/Services/CatalogDataSourceLoader.cs

```diff
@@ -1,129 +1,118 @@
-﻿// Copyright (c) Microsoft Corporation and Contributors
-// Licensed under the MIT license.
-
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
-using DevHome.SetupFlow.Common.Helpers;
-using DevHome.SetupFlow.Models;
-
-namespace DevHome.SetupFlow.Services;
-public class CatalogDataSourceLoader : IDisposable
-{
-    private readonly SemaphoreSlim _lock = new (initialCount: 1, maxCount: 1);
-    private readonly IEnumerable<WinGetPackageDataSource> _dataSources;
-    private readonly Dictionary<WinGetPackageDataSource, IList<PackageCatalog>> _catalogsMap = new ();
-    private bool _disposedValue;
-
-    public CatalogDataSourceLoader(IEnumerable<WinGetPackageDataSource> dataSources)
-    {
-        _dataSources = dataSources;
-    }
-
-    public int CatalogCount => _dataSources.Sum(dataSource => dataSource.CatalogCount);
-
-    public async Task InitializeAsync()
-    {
-        await _lock.WaitAsync();
-        try
-        {
-            foreach (var dataSource in _dataSources)
-            {
-                await InitializeDataSourceAsync(dataSource);
-            }
-        }
-        finally
-        {
-            _lock.Release();
-        }
-    }
-
-    public async IAsyncEnumerable<IList<PackageCatalog>> LoadCatalogsAsync()
-    {
-        await _lock.WaitAsync();
-        try
-        {
-            foreach (var dataSource in _dataSources)
-            {
-                IList<PackageCatalog> dataSourceCatalogs;
```

Comment:
Caching removed from here in favor of a superior caching service (WinGetPackageCache) which caches the packages that were fetched instead of entire catalogs.